docs/cody/capabilities/ignore-context.mdx

```diff
@@ -1,10 +1,151 @@
+
+    By default, there are no restrictions on the repositories Cody can use for context in requests to third-party LLMs.
+    All repositories are included and none are excluded.
+
+2.  **Only `include` rules are specified**
+
+    Cody is restricted to using content from repositories whose names match any of the specified patterns in the include field.
+
+    Since no exclude rules are specified, no repositories are explicitly excluded.
+
+    Note: An include field might contain a catch-all pattern (e.g., a regexp matching any string). In this scenario,
+    Cody can access content from any repository, as a single match in the include rules is sufficient for inclusion.
+
+    Example:
+
+```json
+{
+	"cody.contextFilters": {
+		"include": [
+			{"repoNamePattern": "^github\\.com\\/sourcegraph\\/.+"},
+			{"repoNamePattern": ".*cody.*"}
+		]
+	}
+}
+```
+
+In the example above, in its requests to third-part LLMs Cody can use content from repositories whose names either start with 'github.com/sourcegraph/' or contain 'cody'.
+
+3.  **Only `exclude` rules are specified**
+
+    In the absence of include rules, all repositories are considered included by default.
+
+    Cody is prohibited from using content from any repository whose name matches at least one pattern specified in the exclude field.
+
+    Note: An exclude field may also contain a catch-all condition (e.g., a regexp matching any string).
+    If such a match occurs, Cody is restricted from using content from any repository.
+
+Example:
+
+```json
+{
+	"cody.contextFilters": {
+		"exclude": [
+			{
+				"repoNamePattern": "^github\\.com\\/sourcegraph\\/cody-analytics$"
+			},
+			{
+				"repoNamePattern": ".*secret.*"
+			}
+		]
+	}
+}
+```
+
+In this example, in its requests to third-part LLMs Cody is barred from using repositories named "github.com/sourcegraph/cody-analytics" or those containing the substring "secret".
+
+4. **Both `include` and `exclude` rules are specified**
+
+Cody in requests to third-party LLMs can use content from a repository if its name matches any of the include patterns
+and does not match any of the exclude patterns.
+
+Thus, the exclude rules filter only the repositories allowed by the include rules.
+
+Example:
+
+```json
+{
+	"cody.contextFilters": {
+		"include": [
+			{
+				"repoNamePattern": "^github\\.com\\/sourcegraph\\/.+"
+			}
+		],
+		"exclude": [
+			{
+				"repoNamePattern": "^github\\.com\\/sourcegraph\\/cody-analytics$"
+			},
+			{
+				"repoNamePattern": ".*cody.*"
+			}
+		]
+	}
+}
+```
+
+For the provided configuration, in its requests to third-party LLms Cody can use content from repositories starting with
+"github.com/sourcegraph/" but not from those named "github.com/sourcegraph/cody-analytics" or containing the substring "cody".
+
+## sourcegraph.com
+
+<Callout type="note">
+	This feature is currently in the Experimental stage and only supported with
+	VS Code extension.
+</Callout>
```

Comment:
@MaedahBatool  whether we should have a callout on both sections or just one? Cody Ignore File is going to continue being experimental, while the repo based (Context Filters in the section above) will be GA'd for Cody Enterprises late May with JBs GA.

Reply:
@aramaraju  let's have separate callouts for both. :)